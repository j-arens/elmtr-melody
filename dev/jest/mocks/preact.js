module.exports = {
    unmountComponent: jest.fn(),
};
