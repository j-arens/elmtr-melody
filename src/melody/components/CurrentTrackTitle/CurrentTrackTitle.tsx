import { WithOptionalClassName } from '@components/type';
import { h } from 'preact';

export interface StateProps {
    currentTrackTitle: string;
}

type Props = StateProps & WithOptionalClassName;

export default ({ currentTrackTitle, className = '' }: Props) => (
    <p class={className}>
        {currentTrackTitle}
    </p>
);
