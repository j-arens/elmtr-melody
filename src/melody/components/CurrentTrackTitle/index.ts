import { WithOptionalClassName } from '@components/type';
import { State } from '@redux/type';
import { connect } from 'preact-redux';
import CurrentTrackTitle, { StateProps } from './CurrentTrackTitle';

const mapStateToProps = (state: State) => {
    const { currentTrack, tracks } = state;
    const activeTrack = tracks[currentTrack];
    return {
        currentTrackTitle: activeTrack.media_details.title,
    };
};

export default connect<StateProps, {}, WithOptionalClassName>(mapStateToProps)(CurrentTrackTitle);
