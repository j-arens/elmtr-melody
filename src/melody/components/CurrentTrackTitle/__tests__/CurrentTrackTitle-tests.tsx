import { h } from 'preact';
import { render } from 'preact-render-to-string';
import CurrentTrackTitle from '../CurrentTrackTitle';

describe('CurrentTrackTitle', () => {
    it('renders', () => {
        const component = render(
            <CurrentTrackTitle
                currentTrackTitle="rofl"
                className="purple-silk"
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
