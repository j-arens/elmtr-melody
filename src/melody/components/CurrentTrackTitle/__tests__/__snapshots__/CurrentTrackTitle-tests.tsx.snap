// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentTrackTitle renders 1`] = `<p class="purple-silk" data-melody-title>rofl</p>`;
